src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -920,9 +920,9 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
                 throw new ArgumentNullException(nameof(elementTypes));
             }
 
-            if (elementTypes.Length <= 1)
+            if (elementTypes.Length == 0)
             {
-                throw new ArgumentException(CodeAnalysisResources.TuplesNeedAtLeastTwoElements, nameof(elementNames));
+                throw new ArgumentException(CodeAnalysisResources.TuplesNeedAtLeastOneElement, nameof(elementNames));
```

Comment:
The new message complains about ```elementTypes``` and the ```if``` is checking ```elementTypes```, but the exception uses  ```nameof(elementNames)```.